decompiler/structures/ast/syntaxgraph.py

```diff
@@ -423,7 +423,7 @@ def _substitute_node(self, replacee: AbstractSyntaxTreeNode, replacement: Abstra
         for child in children:
             self._add_edge(replacement, child)
         if isinstance(parent, SeqNode):
-            parent._sorted_children = tuple(replacement if child == replacee else child for child in parent._sorted_children)
+            parent._sorted_children = tuple(replacement if hash(child) == hash(replacee) else child for child in parent._sorted_children)
```

Comment:
I get why there are all these fixes changing `child == replacee` to `hash(child) == hash(replacee)` since you want to use the hash function for comparision, but maybe we should have a helper function for this if its used this extensively.
Not sure about the context, but if it is about identity (e.g. two pointers pointing to the exact same object - without any `copy()` call or something like this, you could also use `child is replacee`.

Reply:
Yes, I want that these are the exact same nodes in the AST. I can change it to `is`.